addons/account/models/account_journal.py

```diff
@@ -1094,3 +1114,52 @@ def _process_reference_for_sale_order(self, order_reference):
         '''
         self.ensure_one()
         return order_reference
+
+    # -------------------------------------------------------------------------
+    # E-Invoice Related Methods
+    # -------------------------------------------------------------------------
+
+    @api.model
+    def _postprocess_received_einvoices(self, moves):
+        journal = moves.mapped('journal_id').ensure_one()
+
+        if not (einvoice_count := len(moves)):
+            return
+
+        journal._send_einvoice_received_mail_if_needed(einvoice_count)
+
+    def _send_einvoice_received_mail_if_needed(self, einvoice_count: int):
+        self.ensure_one()
+        if not all((
+                einvoice_count,
+                self.notify_on_incoming_einvoice,
+                self.incoming_einvoice_notification_email,
+        )):
+            return
+
+        mail_template = self.env.ref('account.mail_template_einvoice_notification', raise_if_not_found=False)
+        if not mail_template:
+            return
+        ctx = {'einvoice_count': einvoice_count}
+        mail_template.with_context(ctx).send_mail(self.id, force_send=True)
+
```

Comment:
```suggestion
        mail_template.with_context(einvoice_count=einvoice_count).send_mail(self.id, force_send=True)
```
Fyi, you should almost never do that.
`with_context(ctx)` shadows completely the current context including the lang of the user.
`with_context(einvoice_count=einvoice_count)` keeps the current context but add the new key in it.

Usually, when you do `with_context(ctx)`, you already copied the whole context in it like:
```py
ctx = dict(self._context)
ctx['einvoice_count'] = einvoice_count
mail_template.with_context(ctx).send_mail(self.id, force_send=True)
```